### PR TITLE
Introduce Coord<D> type

### PR DESCRIPTION
**Pull request by stigrj**
- Define `mrcpp::Coord<D> = std::array<double, D>`
- Replace all `evalf` arguments
- Replace macro `define MWNodeVector` with `using MWNodeVector`

### CODE REVIEW

File: examples/derivative.cpp [hunk 1]
```diff
@@ -34,11 +32,11 @@ int main(int argc, char **argv) {
...
     const auto alpha = 3.0;
     const auto r_0 = mrcpp::pi - 3.0;
-    auto f = [alpha, r_0] (const double *r) -> double {
+    auto f = [alpha, r_0] (const mrcpp::Coord<D> &r) -> double {
         auto R = fabs(r[0] - r_0);
         return exp(-alpha*R);
     };
-    auto df = [alpha, r_0] (const double *r) -> double {
+    auto df = [alpha, r_0] (const mrcpp::Coord<D> &r) -> double {
         auto R = fabs(r[0] - r_0);
```
Comment (robertodr) [inline examples/derivative.cpp]: Nitpick, but replacing `fabs` with `std:abs` will work for any numerical type, not only floats/

File: src/functions/GaussFunc.cpp [hunk 2]
```diff
@@ -324,10 +319,10 @@ double GaussFunc<3>::calcCoulombEnergy(GaussFunc<3> &gf) {
 
     BoysFunction boys(0);
 
-    double boysArg = alpha*Rpq_2;
-    double boysFac = boys.evalf(&boysArg);
+    Coord<1> boysArg = {alpha*Rpq_2};
```
Comment (robertodr) [inline src/functions/GaussFunc.cpp]: Is the `=` needed here when using brace initialization?

File: tests/functions/polynomial.cpp [hunk 3]
```diff
@@ -55,15 +55,15 @@ TEST_CASE("Polynomial evaluation", "[poly_evalf], [polynomials]") {
     Polynomial P(c, &a, &b);
 
     SECTION("Evaluation within bounds") {
-        double x = 1.5;
-        double calc_val = P.evalf(&x);
-        double ref_val = 1.0 + 2.0*x*x;
+        Coord<1> x = {1.5};
```
Comment (robertodr) [inline tests/functions/polynomial.cpp]: I think `Coord<1> x{1.5};` might achieve the same effect.

File: examples/derivative.cpp (hunk 1 above)
Comment (stigrj) [reply examples/derivative.cpp]: Yes, I'm replacing them whenever I see them. I guess I missed this one

File: tests/functions/polynomial.cpp (hunk 3 above)
Comment (stigrj) [reply tests/functions/polynomial.cpp]: Hmm, I found this version better than `x({1.5})` which I had first, but if it works without the `()`, then I agree

Comment (robertodr) [reply tests/functions/polynomial.cpp]: It should work _without_ `()` there are some subtleties for copying _vs_ initializing, but it shouldn't matter at all in this case.

**Comment by stigrj** [comment]
I think it's pretty :unicorn:

**Comment by robertodr** [comment]
Me too :unicorn: Also, it makes clearer what this _means_ rather than how it's implemented, which might help newcomers who know what a coordinate vector is, but not necessarily what a `std::array<double, D>` is.

**Pull request merged**